### ISSUE TITLE
Allow `'` as part of identifiers

### ISSUE THREAD

**Issue by Marwes** [Marwes/gluon#169]
`'` is allowed to be part of identifiers as long as it is not the first character in Haskell. It is usually used to indicate a binding which is related to another binding which lacks the apostrophe.

```f#
let map f xs =
    let map' xs  = ...
    map' xs
map
```

**Comment by Marwes** [comment]
This is the location in the lexer which specifies what identifiers are composed of https://github.com/Marwes/gluon/blob/7cca8f429d22c411b8208d75569e0047826b6c0f/parser/src/lexer.rs#L330-L338.

**Comment by lpelleau** [comment]
Hi!
I've check this issue and (I think) solved it (tests passed, plus a new one I added).
The thing is, I run `cargo run -- -i` then `let a' = 5` and it get me an error : 
```
> let a' = 5
<top>:Line: 1, Column: 10: Unexpected `end of input`
Expected ``in` or an expression in the same column as the `let``

let a' = 5
         ^
```
Is this normal ? (I don't think so, it does the same with `let a = 5` -> no quote)
Thanks

**Comment by Marwes** [comment]
Hi @lpelleau !

That is normal, `let` bindings are required to have `in <expr>` after them to ensure that they always return a value. https://github.com/Marwes/gluon/blob/master/TUTORIAL.md#variable-bindings

**Comment by Marwes** [comment]
Closed by #172
